### PR TITLE
Implement Transfer DLQ entity path helper

### LINKED ISSUE #165: EntityNameHelper doesn't provide a helper for Transfer​ Dead​Letter​ Path
[EntityNameHelper](https://github.com/Azure/azure-service-bus-dotnet/blob/d88ae6a6fd597c8274e6b8112aad4499f5eda74c/src/Microsoft.Azure.ServiceBus/Primitives/EntityNameHelper.cs#L9) doesn't expose [`Format​Transfer​Dead​Letter​Path(string)`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.servicebus.messaging.queueclient.formattransferdeadletterpath?view=azureservicebus-4.0.0#Microsoft_ServiceBus_Messaging_QueueClient_FormatTransferDeadLetterPath_System_String_) method.

### PR DESCRIPTION
**Pull request by SeanFeldman**
Fixes #165 

@jtaubensee please review

### CODE REVIEW

**Comment by jtaubensee** [comment]
@SeanFeldman - I'll bet you wish those were just warnings now, huh? ;)

**Comment by SeanFeldman** [comment]
What I wish is for the errors to bubble up. Can't see what's wrong.

File: src/Microsoft.Azure.ServiceBus/Primitives/EntityNameHelper.cs [hunk 1]
```diff
@@ -8,11 +8,13 @@ namespace Microsoft.Azure.ServiceBus
     /// </summary>
     public static class EntityNameHelper
     {
-        private const string PathDelimiter = @"/";
-        private const string Subscriptions = "Subscriptions";
-        private const string SubQueuePrefix = "$";
-        private const string DeadLetterQueueSuffix = "DeadLetterQueue";
-        private const string DeadLetterQueueName = SubQueuePrefix + DeadLetterQueueSuffix;
+        public const string PathDelimiter = @"/";
```
Comment (SeanFeldman) [inline src/Microsoft.Azure.ServiceBus/Primitives/EntityNameHelper.cs]: No idea what needs to be added for these. BTW, do these even have to be public?

Comment (jtaubensee) [inline src/Microsoft.Azure.ServiceBus/Primitives/EntityNameHelper.cs]: They don't, I made them private in my previous PR: https://github.com/Azure/azure-service-bus-dotnet/pull/137/files#diff-96579f4f545d4098216eff74b430e8f2

Comment (SeanFeldman) [inline src/Microsoft.Azure.ServiceBus/Primitives/EntityNameHelper.cs]: Looks like i've messed that up. Fixing.

**Comment by jtaubensee** [comment]
BTW - `<summary></summary>` is exactly what I was referring to here: https://github.com/Azure/azure-service-bus-dotnet/pull/137#discussion_r118538567

lol

**Comment by SeanFeldman** [comment]
If you noticed, I did comment on PR that I'd need help with those 😉 
Reverting that change. They should stay private.

**Pull request merged**